source/server/converters.h

```diff
@@ -371,15 +372,21 @@ inline CVIAbsoluteTime convert_from_grpc(const google::protobuf::Timestamp& valu
   return cviTime;
 }
 
-// Convert ticks (100ns since Jan 1, 0001) to PrecisionTimestamp
-inline void convert_ticks_to_precision_timestamp(int64 ticks, ::ni::protobuf::types::PrecisionTimestamp* timestamp)
+// Convert .NET/DAQmx ticks (100ns since Jan 1, 0001) to NI-BTF PrecisionTimestamp
+inline void convert_dot_net_daqmx_ticks_to_btf_precision_timestamp(int64 dot_net_ticks, ::ni::protobuf::types::PrecisionTimestamp* timestamp)
 {
-  const double seconds = static_cast<double>(ticks) * SecondsPerTick;
-  const int64 seconds_int = static_cast<int64>(std::floor(seconds));
-  timestamp->set_seconds(seconds_int);
-  const double fractional_seconds = std::abs(seconds - static_cast<double>(seconds_int));
-  const uint64_t fractional_seconds_uint = static_cast<uint64_t>(fractional_seconds * UINT64_MAX);
-  timestamp->set_fractional_seconds(fractional_seconds_uint);
+  const int64 dot_net_ticks_since_1904 = dot_net_ticks - SecondsFromDAQmx0001EpochToCVI1904Epoch * DotNetTicksPerSecond;
+  const double total_seconds = static_cast<double>(dot_net_ticks_since_1904) / DotNetTicksPerSecond;
+  double integer_part;
+  double fractional_part = std::modf(total_seconds, &integer_part);
+  
+  if (fractional_part < 0) {
+    integer_part -= 1;
+    fractional_part += 1;
+  }
```

Comment:
This is incorrect logic for NI-BTF. 

```
    double msb;
    double lsb = modf(-1.25, &msb);
    printf("%f %f\n", msb, lsb);
```

msb=-1.0 lsb=-0.25

You add/subtract to get
msb=-2.0 lsb=0.75

NI-BTF is fixed point, so the fractional part is an extension of the integer part and shares its sign. You are treating the fractional part as if it's always added, regardless of the sign of the integer part. I think struct timeval might work this way, but NI-BTF doesn't.

A fixed point number of msb=-2.0 lsb=0.75 has an absolute value of 2.75. The correct value would be msb=-1.0 lsb=0.25.